plugins/thunderbolt/fu-thunderbolt-image.c

```diff
@@ -491,24 +491,77 @@ get_host_locations (guint16 id)
 	}
 }
 
+/*
+ * Finds optional multi controller (MC) entry from controller DROM.
+ * Returns TRUE if the controller did not have MC entry or the
+ * controller and image MC entries match. In any other case FALSE is
+ * returned and error is set accordingly.
+ */
+static gboolean
+compare_device_mc (const FuThunderboltFwObject *fw,
+		   const FuThunderboltFwObject *image,
+		   GError **error)
+{
+	FuThunderboltFwLocation image_mc_loc = { .section = DROM_SECTION, .description = "Multi Controller" };
+	FuThunderboltFwLocation fw_mc_loc = { .section = DROM_SECTION, .description = "Multi Controller" };
+	g_autoptr(GByteArray) image_mc = NULL;
+	g_autoptr(GByteArray) fw_mc = NULL;
+
+	if (!read_drom_entry_location (fw, DROM_ENTRY_MC, &fw_mc_loc, error))
+		return FALSE;
+
+	/* it is fine if the controller does not have MC entry */
+	if (fw_mc_loc.offset == 0)
+		return TRUE;
+
+	if (!read_drom_entry_location (image, DROM_ENTRY_MC, &image_mc_loc, error))
+		return FALSE;
+
+	if (image_mc_loc.offset == 0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE,
+				     "The FW image file misses multi controller entry");
```

Comment:
drop the `The` -- perhaps something like `firmware does not have multi-controller entry point`